File: 3-LongestSubstringWithoutRepeatingCharacters/3-LongestSubstringWithoutRepeatingCharacters.java
// Last updated: 7/14/2026, 1:55:03 PM
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int left=0;
        int max_len=0;
        Set<Character> set=new HashSet<>();
        for(int right=0;right<s.length();right++){
            char c=s.charAt(right);
            while(set.contains(c)){
                set.remove(s.charAt(left));
                left++;
            }
            set.add(c);
            int current_len=right-left+1;
            max_len=Math.max(max_len,current_len);
        }
        return max_len;
    }
}

File: 3-LongestSubstringWithoutRepeatingCharacters/3-LongestSubstringWithoutRepeatingCharacters.py
# Last updated: 9/8/2026, 1:42:07 PM
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        max_len=0
        left=0
        s1=set()
        for right in range(len(s)):
            while s[right] in s1:
                s1.remove(s[left])
                left+=1
            s1.add(s[right])
            c_len=len(s1)
            max_len=max(max_len,c_len)
        return max_len